examples/react/react-app/src/App.tsx

```diff
@@ -23,25 +22,38 @@ function App() {
         isError: false,
     });
 
-    // get current component values
-    const position = dojo_starter.position.find({
-        player: account?.account.address,
-    });
-    const moves = dojo_starter.moves.find({ player: account?.account.address });
+    // Get current player entity
+    // This will return a player with the models that you specified
+    // {
+    //     player: string;
+    //     remaining: number;
+    //     last_direction: Direction;
+    //     vec: Vec2;
+    // }
+    const player = dojo_starter.findEntity<[MovesModel, PositionModel]>([
```

Comment:
This is nice. Would we be able to type annotations during execution? What would this look like in vanilla js?

Reply:
Not sure what you mean by "during execution"?

Vanilla JS won't be able to have any type interference, so it would be 

```js
const player = dojo_starter.findEntity([
    { player: account.account.address },
    {},
]);
```

where both the arguments and the return value won't be typed at all.

Reply:
in that case, how would the call know what models to lookup?

Reply:
Oh boy, I was so focused on the typing only that I totally forgot to think about actual implementation..

So again updated the code to now work as expected.

The example is now

```ts
    const [moves, position] = dojo_starter.findEntity([
        { model: "Moves", query: { remaining: 5 } },
        { model: "Position" },
    ] as const);
 ```

So in JS it would just work without defining the `as const`!

Reply:
This is looking nice!

Do you have thoughts on supporting logical (AND, OR, NOT) and comparison (>, <, =, etc) operators?

For example, in the case above, `remaining >= 5`.

We could take some inspiration from Bevy here: https://bevy-cheatbook.github.io/programming/queries.html#query-filters

Generally though, it very quickly starts to look similar to the grpc api: https://github.com/dojoengine/dojo/blob/953e5e5801990ed6857a08e02e625d09930cc106/crates/torii/grpc/proto/types.proto#L93

I still wonder if that should be a wrapper and the low level should basically just wrap the grpc api and these nicer abstractions should be separate

Reply:
The operators are already part of this RFC! There's [an example included in the commented code](https://github.com/dojoengine/dojo.js/pull/160/files#diff-155fdc5dcfcb1004c2286abf68a3ff90984dcf6d2ee167bd0abe0e72d6670998R41).

With regards to it being a low level wrapper, I think this PR still keeps it as a low level wrapper. I'm not sure where you feel abstractions are being made?